### ISSUE TITLE
Need override functionality to editor-templates/templates.php 

### ISSUE THREAD

**Issue by prasidhda** [pojome/elementor#3875]
I am currently trying to build the add-on plugin which requires to customize editor-templates/templates.php  file but i don't see any filter hooks to override with my add-on plugin. Is there any way to override that particular file?

**Comment by bainternet** [comment]
@prasidhda 

Nope, what are you trying to do? maybe there is a different way to achieve it.

**Comment by prasidhda** [comment]
I am trying to add new tabs alongside "Predesiged Templates" and "My templates" like shown in the following sceenshot then allow the functionality of inserting custom templates from there. 
![override-templates-php](https://user-images.githubusercontent.com/11455326/37872695-2a934a38-302d-11e8-8eb2-8470095633e9.png)
We can add by JavaScript DOM manipulation but it would be better if the main plugin provides the flexibility of adding custom section. We can take the example of Woo commerce which literally provides the flexibility to override every templates and texts.

**Comment by norewp** [comment]
This is a much needed feature by developers and I therefore second it with a +1. Please consider allowing native hooking rather than JS hacks :)

**Comment by tristanleboss** [comment]
After 1 month of trial of Elementor Pro, I have to admit it lacks love on the developper side. The control we have on it is really light compared to the potential of the tool.

There is a few big things like this one that you can't do and would be terribly useful to really blend Elementor in a project. Nowadays, the feeling I have is that Elementor goes on top of it without really blending in it.

The ability to save all the styles of a block to use it on another one is also a blocker. Of course, you can use CSS classes but you loose the Style tab along with the ease of use (1 click) for non-techie users.

For techie users, it should be possible to open a popup which shows the currently generated CSS for a block with the ability to edit it. Once updated, the Style tab should reflect the chqnges. Indeed, for a CSS guru, using the Style tab UI is a big loose of time. 

Same thing for the ability to create containers blocks (a block that can conatins blocks inside of it) or multiple content area on one page.
